tyk-pro/scripts/bootstrap_k8s.sh

```diff
@@ -1,28 +1,176 @@
+}
+
+curl_message() {
+  echo 'ERROR: Please install cURL'
+  exit 1
+}
+
+generate_credentials(){
+  error=0
+  [[ -z "$1" ]] && return "6" || echo ""
+  error="$?"
+  [[ $error != 0 ]] && return $error
+  echo "Creating Organisation"
+  ORGDATA=$(create_organisation)
+  error="$?"
+  [[ $error != 0 ]] && return $error || echo "ORG DATA: $ORGDATA"
+  export ORGID=$(org_id "$ORGDATA")
+  error="$?"
+  [[ $? != 0 ]] && return $error || echo "ORG ID: $ORGID"
+  echo ""
+  echo "Adding new user"
+  export USER_AUTH_CODE=$(create_user $ORGID)
+  error=$?
+}
+
+exit_reason(){
+  case "$1" in
+    "3")
+      echo "ERROR: Unable to connect to the Dashboard. Please check that you are using the correct hostname."
+      exit $1
+      ;;
+    "4")
+      echo "ERROR: Unable to parse JSON"
+      exit $1
+      ;;
+    "5")
+      echo "Unexpected error"
+      exit $1
+      ;;
+    "6")
+      echo "ERROR: Hostname should not contain port number or protocol"
+      ;;
+    *)
+      echo ""
+      ;;
+  esac
+}
+
+if_present_echo(){
+  [[ -z "$1" ]] && return $2 || echo "$1"
+}
+
+create_organisation() {
+  ORGDATA=$(curl --silent --header "admin-auth: $TYK_ADMIN_SECRET" --header "Content-Type:application/json" --data '{"owner_name": "'"$TYK_ORG_NAME"'", "cname_enabled": true, "cname": "'$TYK_ORG_CNAME'"}' $DASHBOARD_HOSTNAME/admin/organisations 2>&1)
+  if_present_echo "$ORGDATA" 3
+}
+
+org_id(){
+  ORGID=$(echo $1 | jq -r '.Meta')
+  if_present_echo $ORGID 4
+}
+
+create_user(){
+  USER_DATA=$(user_data $1)
+  AUTH_CODE=$(echo $USER_DATA | jq -r '.Message')
+  USER_ID=$(echo $USER_DATA | jq -r '.Meta.id')
+  ## set user password
+  response=$(curl --silent --header "authorization: $AUTH_CODE" --header "Content-Type:application/json" $DASHBOARD_HOSTNAME/api/users/$USER_ID/actions/reset --data '{"new_password":"'$TYK_ADMIN_PASSWORD'", "user_permissions": { "IsAdmin": "admin" }}')
+  if_present_echo "$AUTH_CODE" 4
+}
+
+user_data(){
+  user_curl=$(curl --silent --header "admin-auth: $TYK_ADMIN_SECRET" --header "Content-Type:application/json" --data '{"first_name": "'$TYK_ADMIN_FIRST_NAME'", "last_name": "'$TYK_ADMIN_LAST_NAME'", "email_address": "'$TYK_ADMIN_EMAIL'", "active": true, "org_id": "'$1'", "user_permissions": { "IsAdmin": "admin" }}' $DASHBOARD_HOSTNAME/admin/users 2>&1)
+  if_present_echo $user_curl 3
+}
+
+main $DASHBOARD_HOSTNAME
 
 kubectl create secret -n ${NAMESPACE} generic tyk-operator-conf \
-  --from-literal "TYK_AUTH=${TYK_AUTH}" \
-  --from-literal "TYK_ORG=${TYK_ORG}" \
-  --from-literal "TYK_MODE=${TYK_MODE}" \
-  --from-literal "TYK_URL=${TYK_URL}"
-
-kubectl create secret -n ${NAMESPACE} generic tyk-login-details \
-  --from-literal "TYK_USER=${TYK_USER}" \
-  --from-literal "TYK_PASS=${TYK_PASS}"
+  --from-literal "TYK_AUTH=${USER_AUTH_CODE}" \
+  --from-literal "TYK_ORG=${ORGID}" \
+  --from-literal "TYK_MODE=pro" \
+  --from-literal "TYK_URL=http://${DASHBOARD_HOSTNAME}"
+
+if [ "$BOOTSTRAP_PORTAL" = "true" ]; then
```

Comment:
I don't think there's a need to put the function definitions under the conditionals.

Also, would be good to refactor this a little so that all the necessary bootstrapping happens in the `main()` function, where the portal parts are under the conditional.